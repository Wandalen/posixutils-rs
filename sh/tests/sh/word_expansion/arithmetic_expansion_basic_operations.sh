echo $(((22)))
echo $((+1))
echo $((-1))
echo $((~1))
echo $((!0))
echo $((2 * 4))
echo $((4 / 2))
echo $((5 / 2))
echo $((5 % 2))
echo $((22 + 4))
echo $((22 - 4))
echo $((2 << 2))
echo $((8 >> 2))
echo $((2 < 4))
echo $((2 <= 3))
echo $((2 > 4))
echo $((2 >= 3))
echo $((2 == 2))
echo $((2 != 2))
echo $((2 & 3))
echo $((2 ^ 3))
echo $((3 | 4))
echo $((1 && 0))
echo $((1 || 0))
echo $((1 ? 2 : 3))
echo $((0 ? 2 : 3))
x=1
echo $((x = 2))
echo $x
echo $((x += 3))
echo $x
echo $((x -= 2))
echo $x
echo $((x *= 4))
echo $x
echo $((x /= 2))
echo $x
echo $((x %= 3))
echo $x
echo $((x <<= 2))
echo $x
echo $((x >>= 1))
echo $x
echo $((x &= 3))
echo $x
echo $((x ^= 3))
echo $x
echo $((x |= 4))
echo $x
