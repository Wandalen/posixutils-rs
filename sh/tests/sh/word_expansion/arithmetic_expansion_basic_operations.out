22
1
-1
-2
1
8
2
2
1
26
18
8
2
1
1
0
0
1
0
2
1
7
0
1
2
3
2
2
5
5
3
3
12
12
6
6
0
0
0
0
0
0
0
0
3
3
7
7
